### ISSUE TITLE
Producer not working in .NET Framework clients when using Unity DI

### ISSUE THREAD

**Issue by ravi-kotha** [Farfetch/kafka-flow#223]
When trying to use a injected typed producer into the application, we get an error 

There is no instance container registered with the given ID. It's a bug! Please, open an issue!\r\nParameter name: InstanceContainerId\r\nActual value was f82cdfd2-5da7-4553-8c41-18c4601d11a9."}    System.Exception {System.ArgumentOutOfRangeException}

The error occurs in KafkaFlow.MiddlewareExecutor.CreateInstance

We did not have this issue in the version 1.5.4, but when we switched to the latest version we started seeing this issue.

**Comment by brenden-sharp** [comment]
I work with Ravi and dug into this this morning and this is what I found and shared with him.

So the problem is that when KafkaFlow is trying to register the MiddlewareInstanceContainers for the SerializerProducerMiddleware it's doing it wrong for Unity.  If you register multiple things of the same type in .Net Core (using IServiceProvider) it will keep multiple instances registered, which is why they call ResolveAll later to get them all then look for one with a specific ID.  Unity doesn't work that way.  If you register the same type like that multiple times and don't name them, it just updates the original registration.  On top of that, ResolveAll in Unity only returns named registrations, not the default, unnamed registration.  So it looks like those registrations need to be named to work with Unity.  I tried adding naming to the appropriate method in UnityDependencyConfigurator but that broke other things (our other types couldn't get the default registration for a producer and failed to resolve).  Looks like there may need to be another method added there to allow multiple registrations of certain types when using unity and name them something random.  It just can't be used for everything or it breaks other stuff.
